src/dmd/cparse.d

```diff
@@ -1442,6 +1443,30 @@ final class CParser(AST) : Parser!AST
                 }
             }
 
+            /* GNU Extensions
+             * init-declarator:
+             *    declarator simple-asm-expr[opt] gnu-attributes[opt]
+             *    declarator simple-asm-expr[opt] gnu-attributes[opt] = initializer
```

Comment:
I've been using `(opt)` elsewhere in the grammar comments.

A link to the GNU documentation here would be a plus.

Reply:
What is the grammar for `simple-asm-expr` and `gnu-attributes` ?

Reply:
`gnu-attributes` is covered by `cparseAttribute`'s documentation of `attribute-specifier`.   However, C2X uses `attribute-specifier` for the new-style attribute syntax:
```
attribute-specifier-sequence:
  attribute-specifier-sequence[opt] attribute-specifier

attribute-specifier:
  [ [ attribute-list ] ]

attribute-list:
  attribute[opt]
  attribute-list, attribute[opt]

attribute:
  attribute-token attribute-argument-clause[opt]

attribute-token:
  standard-attribute
  attribute-prefixed-token

standard-attribute:
  identifier

attribute-prefixed-token:
  attribute-prefix :: identifier

attribute-prefix:
  identifier

attribute-argument-clause:
  ( balanced-token-sequence[opt] )
```

So the documentation in cparse.d should be altered to be clear that `cparseAttribute` really is `cparseGnuAttribute`.

Reply:
`simple-asm-expr` is covered by documentation for `cparseAsmLabel`, and it has a link to GCC documentation.

Reply:
#12556 and #12557